### PR TITLE
Fix Constructed Inventory colliding with defaults on controller_inventory_sources

### PR DESCRIPTION
**Pull request by Klaas-**
# What does this PR do?
Follow up to #720 references #601

this changes the behaviour to allow modifications in constructed inventories, for example if you want to set a limit or modify source_vars.

I've updated the test constructed inventory to use both features.

# How should this be tested?
Constructed inventory in [tests/configs/inventory_sources.yml](tests/configs/inventory_sources.yml) should work

# Is there a relevant Issue open for this?

references #601 

# Other Relevant info, PRs, etc
#720

### CODE REVIEW

**Review by sean-m-sullivan — CHANGES_REQUESTED** [review]
Need to get the rest of it fixed, so it passes PR's, add example, changelog fragment, It looks to be trying to sync the project or get the project, so that may be the issue.

File: roles/inventory_sources/tasks/main.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@
     description:                          "{{ __controller_source_item.description | default(('' if controller_configuration_inventory_sources_enforce_defaults else omit), true) }}"
     inventory:                            "{{ __controller_source_item.inventory.name | default(__controller_source_item.inventory) | mandatory }}"
     organization:                         "{{ __controller_source_item.inventory.organization.name | default(__controller_source_item.organization | default(('' if controller_configuration_inventory_sources_enforce_defaults else omit), true)) }}"
-    source:                               "{{ __controller_source_item.source | default(('scm' if controller_configuration_inventory_sources_enforce_defaults else omit), true) }}"
+    source:                               "{{ omit if __controller_source_item.source is defined and __controller_source_item.source == 'constructed' else __controller_source_item.source | default(('scm' if controller_configuration_inventory_sources_enforce_defaults else omit), true) }}"
```
Comment (ivarmu) [inline roles/inventory_sources/tasks/main.yml:9]: The `source` of an auto-generated inventory source for constructed inventories must be `constructed`. 

In the opposite, `source_project` should be `null` (omit).

Here's a complete example from the API:
```json
{
    "count": 1,
    "next": null,
    "previous": null,
    "results": [
        {
            "id": 51,
            "type": "inventory_source",
            "url": "[/api/v2/inventory_sources/51/](https://aap-ha-controller.iam.lab/api/v2/inventory_sources/51/)",
            "related": {
                "created_by": "[/api/v2/users/1/](https://aap-ha-controller.iam.lab/api/v2/users/1/)",
                "modified_by": "[/api/v2/users/1/](https://aap-ha-controller.iam.lab/api/v2/users/1/)",
                "last_job": "[/api/v2/inventory_updates/422/](https://aap-ha-controller.iam.lab/api/v2/inventory_updates/422/)",
                "update": "[/api/v2/inventory_sources/51/update/](https://aap-ha-controller.iam.lab/api/v2/inventory_sources/51/update/)",
                "inventory_updates": "[/api/v2/inventory_sources/51/inventory_updates/](https://aap-ha-controller.iam.lab/api/v2/inventory_sources/51/inventory_updates/)",
                "schedules": "[/api/v2/inventory_sources/51/schedules/](https://aap-ha-controller.iam.lab/api/v2/inventory_sources/51/schedules/)",
                "activity_stream": "[/api/v2/inventory_sources/51/activity_stream/](https://aap-ha-controller.iam.lab/api/v2/inventory_sources/51/activity_stream/)",
                "hosts": "[/api/v2/inventory_sources/51/hosts/](https://aap-ha-controller.iam.lab/api/v2/inventory_sources/51/hosts/)",
                "groups": "[/api/v2/inventory_sources/51/groups/](https://aap-ha-controller.iam.lab/api/v2/inventory_sources/51/groups/)",
                "notification_templates_started": "[/api/v2/inventory_sources/51/notification_templates_started/](https://aap-ha-controller.iam.lab/api/v2/inventory_sources/51/notification_templates_started/)",
                "notification_templates_success": "[/api/v2/inventory_sources/51/notification_templates_success/](https://aap-ha-controller.iam.lab/api/v2/inventory_sources/51/notification_templates_success/)",
                "notification_templates_error": "[/api/v2/inventory_sources/51/notification_templates_error/](https://aap-ha-controller.iam.lab/api/v2/inventory_sources/51/notification_templates_error/)",
                "inventory": "[/api/v2/inventories/27/](https://aap-ha-controller.iam.lab/api/v2/inventories/27/)",
                "last_update": "[/api/v2/inventory_updates/422/](https://aap-ha-controller.iam.lab/api/v2/inventory_updates/422/)"
            },
            "summary_fields": {
                "organization": {
                    "id": 36,
                    "name": "Default",
                    "description": ""
                },
                "inventory": {
                    "id": 27,
                    "name": "test_constructed",
                    "description": "",
                    "has_active_failures": false,
                    "total_hosts": 0,
                    "hosts_with_active_failures": 0,
                    "total_groups": 0,
                    "has_inventory_sources": false,
                    "total_inventory_sources": 0,
                    "inventory_sources_with_failures": 0,
                    "organization_id": 36,
                    "kind": "constructed"
                },
                "last_job": {
                    "id": 422,
                    "name": "test_constructed - Auto-created source for: test_constructed",
                    "description": "",
                    "finished": "2024-01-04T09:02:37.642493Z",
                    "status": "successful",
                    "failed": false,
                    "license_error": false
                },
                "last_update": {
                    "id": 422,
                    "name": "test_constructed - Auto-created source for: test_constructed",
                    "description": "",
                    "status": "successful",
                    "failed": false,
                    "license_error": false
                },
                "created_by": {
                    "id": 1,
                    "username": "admin",
                    "first_name": "",
                    "last_name": ""
                },
                "modified_by": {
                    "id": 1,
                    "username": "admin",
                    "first_name": "",
                    "last_name": ""
                },
                "user_capabilities": {
                    "edit": true,
                    "delete": true,
                    "start": true,
                    "schedule": true
                },
                "credentials": [],
                "resolved_environment": {
                    "id": 4,
                    "name": "Default execution environment",
                    "description": "",
                    "image": "registry.redhat.io/ansible-automation-platform-24/ee-supported-rhel8:latest"
                }
            },
            "created": "2024-01-04T08:55:08.364134Z",
            "modified": "2024-01-04T08:55:08.364140Z",
            "name": "Auto-created source for: test_constructed",
            "description": "",
            "source": "constructed",
            "source_path": "",
            "source_vars": "",
            "scm_branch": "",
            "credential": null,
            "enabled_var": "",
            "enabled_value": "",
            "host_filter": "",
            "overwrite": true,
            "overwrite_vars": true,
            "custom_virtualenv": null,
            "timeout": 0,
            "verbosity": 1,
            "limit": "",
            "last_job_run": "2024-01-04T09:02:37.642493Z",
            "last_job_failed": false,
            "next_job_run": null,
            "status": "successful",
            "execution_environment": null,
            "inventory": 27,
            "update_on_launch": true,
            "update_cache_timeout": 0,
            "source_project": null,
            "last_update_failed": false,
            "last_updated": "2024-01-04T09:02:37.642493Z"
        }
    ]
}
```

**Review by ivarmu — CHANGES_REQUESTED** [review]

**Comment by mr-oz-09** [comment]
Hello, Will this be merged soon? We are waiting on this fix.

**Comment by adrienmrgn** [comment]
I think it miss the integration test, because the inventory source definition for the constructed inventory has no organisation defined, so fall back to Default, whereas the related inventory has moved to Satellite org

**Comment by djdanielsson** [comment]
please update this PR to fix the conflicts and any other requested changes or I will close it soon because it has just been sitting here

**Comment by djdanielsson** [comment]
closing due to this being open for so long and no movement

**Pull request closed without merging**